src/components/BotAppearance/BotAppearance.tsx

```diff
@@ -1,47 +1,56 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import { useAsyncEffect } from '@dwarvesf/react-hooks'
 import { zodResolver } from '@hookform/resolvers/zod'
-import { useToast } from '@mochi-ui/core'
+import { Skeleton, useToast } from '@mochi-ui/core'
 import { useParams } from 'next/navigation'
-import { useCallback, useEffect, useRef } from 'react'
+import { useCallback, useEffect, useRef, useState } from 'react'
 import { FormProvider, useForm } from 'react-hook-form'
 import { z } from 'zod'
 import { api } from '~/utils/api'
 import { SaveBar } from '../SaveBar'
+import { BotAvatarWidget } from './BotAvatarWidget'
+import { BotCompanyLogo } from './BotCompanyLogo'
+import { ColorPicker } from './ColorPicker'
 import { WidgetMessage } from './WidgetMessage'
 import { WidgetName } from './WidgetName'
 
 export interface BotAppearance {
   botId: string
+  companyLogoAttachmentId: string
+  botAvatarAttachmentId: string
   widgetName: string
   widgetSubheading: string
   widgetPlaceholder: string
   widgetWelcomeMsg: string
+  accentColour: string
 }
 
 const schema = z.object({
   botId: z.string(),
+  companyLogoAttachmentId: z.string(),
+  botAvatarAttachmentId: z.string(),
   widgetName: z
     .string()
     .min(1, 'Required')
     .max(50, 'Max length is 50 characters.'),
   widgetSubheading: z.string().max(50, 'Max length is 50 characters.'),
   widgetPlaceholder: z.string().max(100, 'Max length is 100 characters.'),
   widgetWelcomeMsg: z.string().max(100, 'Max length is 100 characters.'),
+  accentColour: z.string().max(7, 'Required'),
 })
 
 export const BotAppearancePage = () => {
   const id = useParams()?.id
   const isInitialData = useRef(false)
   const { toast } = useToast()
-
+  const [isFetchingData, setIsFetchingData] = useState<boolean>(false)
```

Comment:
No need to explicitly declare the type; TypeScript infers it from the default value.
```suggestion
  const [isFetchingData, setIsFetchingData] = useState(false)
```